Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h

```diff
@@ -1704,28 +1704,21 @@ namespace internal {
         return true;
 
       bool done = false;
+
       while(!degenerate_faces.empty())
       {
         halfedge_descriptor h = *(degenerate_faces.begin());
         degenerate_faces.erase(degenerate_faces.begin());
 
-        if (!is_degenerate_triangle_face(face(h, mesh_), mesh_,
-                                         parameters::vertex_point_map(vpmap_)
-                                                    .geom_traits(gt_)))
-          //this can happen when flipping h has consequences further in the mesh
-          continue;
-
-        //check that opposite is not also degenerate
-        degenerate_faces.erase(opposite(h, mesh_));
-
         if(is_border(h, mesh_))
```

Comment:
@janetournois  in such a case, you can simply remove the face. Basically the opposite of [`add_face_to_border()`](https://doc.cgal.org/latest/BGL/group__PkgBGLEulerOperations.html#ga8666be89a7ffa73a5fe82b6b2bdc6f19)

Reply:
`Euler::remove_face()` should be safe (could be a face with 2 border halfedges)